Support for SpeedyBee F405 v3 (#22075)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/speedybeef4v3/README.md

# SpeedyBee F405 v3 Flight Controller

The SpeedyBee F405 v3 is a flight controller produced by [SpeedyBee](http://www.speedybee.com/).

## Features

 - STM32F405 microcontroller
 - BMI270 IMU
 - DPS280 barometer
 - SDCard
 - AT7456E OSD
 - 6 UARTs
 - 9 PWM outputs

## Pinout

![SpeedyBee F405 v3](SpeedyBee_F405_v3_Board_Top.JPG "SpeedyBee F405 v3")
![SpeedyBee F405 v3](SpeedyBee_F405_v3_Board_Bottom.JPG "SpeedyBee F405 v3")

## UART Mapping

The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
receive pin for UARTn. The Tn pin is the transmit pin for UARTn.

 - SERIAL0 -> USB
 - SERIAL1 -> UART1 (DJI-VTX, DMA-enabled)
 - SERIAL2 -> UART2 (RCIN, DMA-enabled) 
 - SERIAL3 -> UART3 (CAM)
 - SERIAL4 -> UART4 (connected to internal BT module)
 - SERIAL5 -> UART5 (ESC Telemetry)
 - SERIAL6 -> UART6 (GPS, DMA-enabled)

## RC Input

> RC input is configured on the R2 (UART2_RX) pin. It supports all RC

Author: Ignored, because must use SBUS pin for SBUS

> RC input is configured on the R2 (UART2_RX) pin. It supports all RC

Reviewer: still not right...
```suggestion
RC input is configured on the R2 (UART2_RX) pin for most RC unidirectioinal protocols except SBUS which should be applied at the SBUS pin. PPM is not supported.
```

protocols except PPM. For protocols requiring half-duplex serial to transmit

> telemetry (such as FPort) you should configure SERIAL2 with half-duplex, pin-swap

Reviewer: ```suggestion
```

> and inversion enabled.

Reviewer: ```suggestion
For CRSF/ELRS/SRXL2 connection of the receiver to T2 will also be required.
```

 
## FrSky Telemetry
 
FrSky Telemetry is supported using the Tx pin of any UART including SERIAL2/UART2. You need to set the following parameters to enable support for FrSky S.PORT (example shows SERIAL2).
 
  - SERIAL2_PROTOCOL 10
  - SERIAL2_OPTIONS 7
  
## OSD Support

The SpeedyBee F405 v3 supports OSD using OSD_TYPE 1 (MAX7456 driver).

## VTX Support

The JST-GH-6P connector supports a standard DJI HD VTX connection. Pin 1 of the connector is 9v so be careful not to connect

> this to a peripheral requiring 5v.

Author: Can only do SBUS on SBUS pin

## PWM Output

The SpeedyBee F405 v3 supports up to 9 PWM outputs. The pads for motor output
M1 to M4 on the motor connector, plus M9 for LED strip or another
PWM output.

The PWM is in 3 groups:

 - PWM 1-4 in group1
 - PWM 5-8 in group2
 - PWM 9 in group3

Channels within the same group need to use the same output rate. If
any channel in a group uses DShot then all channels in the group need
to use DShot. Channels 1-4 support bi-directional DShot.

## Battery Monitoring

> The board has a builtin voltage and current sensor. The current

Reviewer: ```suggestion
The board has a internal voltage sensor and connections on the ESC connector for an external current sensor input.
```

> sensor can read up to 130 Amps. The voltage sensor can handle up to 6S

Reviewer: ```suggestion
The voltage sensor can handle up to 6S.
```

LiPo batteries.

> The correct battery setting parameters are:

Reviewer: ```suggestion
The default battery parameters are:
```

 - BATT_MONITOR 4
 - BATT_VOLT_PIN 10
 - BATT_CURR_PIN 11
 - BATT_VOLT_MULT 11.2

>  - BATT_AMP_PERVLT 52.7

Reviewer: ```suggestion
 - BATT_AMP_PERVLT 52.7 (will need to be adjusted for whichever current sensor is attached)
```
